controllers/cluster_controller.go

```diff
@@ -611,3 +615,17 @@ func (r *ClusterReconciler) setDefaultBundlesRefOrEksaVersion(ctx context.Contex
 	clus.Status.FailureMessage = ptr.String("Management cluster must have either EksaVersion or BundlesRef")
 	return fmt.Errorf("could not set default values")
 }
+
+func validateEksaRelease(ctx context.Context, client client.Client, cluster *anywherev1.Cluster) error {
+	if cluster.Spec.EksaVersion == nil {
+		return nil
+	}
+	if err := validations.ValidateEksaReleaseExistOnManagement(ctx, clientutil.NewKubeClient(client), cluster); err != nil {
```

Comment:
I think you need to distinguish between error reading and the release not existing. You don't want to set an error in the status for the first case.